modules/xfem/src/base/XFEM.C

```diff
@@ -882,6 +929,148 @@ XFEM::initCutIntersectionEdge(
+                e1node != e2node) // This would happen at the crack tip
+            {
+              elem1->set_node(in) = e2node;
+              nodes_to_delete.insert(e1node);
+            }
+            else if (e1node != e2node)
+              nodes_to_delete.insert(e2node);
+          }
+        }
+        else
+          mooseError("Could not find XFEMCutElem for element to be kept in healing");
+
+        if (_displaced_mesh)
+        {
+          Elem * elem1_displaced = _displaced_mesh->elem(it->first->id());
+          Elem * elem2_displaced = _displaced_mesh->elem(it->second->id());
+
+          std::map<unique_id_type, XFEMCutElem *>::iterator cemit =
+              _cut_elem_map.find(elem1_displaced->unique_id());
+          if (cemit != _cut_elem_map.end())
+          {
+            const XFEMCutElem * xfce = cemit->second;
+
+            for (unsigned int in = 0; in < elem1_displaced->n_nodes(); ++in)
+            {
+              Node * e1node_displaced = elem1_displaced->get_node(in);
+              Node * e2node_displaced = elem2_displaced->get_node(in);
+              if (!xfce->isPointPhysical(*e1node_displaced) &&
+                  e1node_displaced != e2node_displaced) // This would happen at the crack tip
+              {
+                elem1_displaced->set_node(in) = e2node_displaced;
+                nodes_to_delete_displaced.insert(e1node_displaced);
+              }
+              else if (e1node_displaced != e2node_displaced)
+                nodes_to_delete_displaced.insert(e2node_displaced);
+            }
+          }
+          else
+            mooseError("Could not find XFEMCutElem for element to be kept in healing");
+
+          elem2_displaced->nullify_neighbors();
+          _displaced_mesh->boundary_info->remove(elem2_displaced);
+          _displaced_mesh->delete_elem(elem2_displaced);
+        }
+
+        cutelems_to_delete.insert(elem2->unique_id());
+        elem2->nullify_neighbors();
+        _mesh->boundary_info->remove(elem2);
+        unsigned int deleted_elem_id = elem2->id();
+        _mesh->delete_elem(elem2);
+        _console << "XFEM healing deleted element: " << deleted_elem_id << "\n";
+        mesh_changed = true;
+      }
+    }
+  }
+
+  for (std::set<Node *>::iterator sit = nodes_to_delete.begin(); sit != nodes_to_delete.end();
+       ++sit)
+  {
+    Node * node_to_delete = *sit;
+    unsigned int deleted_node_id = node_to_delete->id();
+    _mesh->boundary_info->remove(node_to_delete);
+    _mesh->delete_node(node_to_delete);
+    _console << "XFEM healing deleted node: " << deleted_node_id << "\n";
+  }
+
+  if (_displaced_mesh)
+  {
+    for (std::set<Node *>::iterator sit = nodes_to_delete_displaced.begin();
+         sit != nodes_to_delete_displaced.end();
+         ++sit)
+    {
+      Node * node_to_delete = *sit;
+      _displaced_mesh->boundary_info->remove(node_to_delete);
+      _displaced_mesh->delete_node(node_to_delete);
+    }
+  }
+  _console << std::flush;
```

Comment:
You didn't output to console in the loop above, why not move this flush up to the last place where you wrote to the console?

Reply:
did